backend/src/main/kotlin/com/group7/artshare/controller/UserInteractionController.kt

```diff
@@ -35,4 +39,126 @@ class UserInteractionController(
+                return searchService.searchPhysicalExhibition(keywords)
+
+            } ?: return searchService.searchPhysicalExhibition(keywords) //if we want to customize the search results according to user, replace the method call here
+        } catch (e: Exception) {
+            if (e.message == "Invalid token") {
+                throw ResponseStatusException(HttpStatus.UNAUTHORIZED, e.message)
+            } else {
+                throw ResponseStatusException(HttpStatus.BAD_REQUEST, e.message)
+            }
+        }
+    }
+
+    @GetMapping("search_online_gallery")
+    fun searchOnlineGallery(
+        @RequestParam(value = "keywords") keywords: List<String>,
+        @RequestHeader(
+            value = "Authorization",
+            required = false
+        ) authorizationHeader: String?
+    ): List<OnlineGalleryDTO> {
+        try {
+            authorizationHeader?.let {
+                val user =
+                    jwtService.getUserFromAuthorizationHeader(authorizationHeader) ?: throw Exception("Invalid token")
+                return searchService.searchOnlineGallery(keywords)
+
+            } ?: return searchService.searchOnlineGallery(keywords) //if we want to customize the search results according to user, replace the method call here
+        } catch (e: Exception) {
+            if (e.message == "Invalid token") {
+                throw ResponseStatusException(HttpStatus.UNAUTHORIZED, e.message)
+            } else {
+                throw ResponseStatusException(HttpStatus.BAD_REQUEST, e.message)
+            }
+        }
+    }
+
+    @GetMapping("search_discussion_post")
+    fun searchDiscussionPost(
+        @RequestParam(value = "keywords") keywords: List<String>,
+        @RequestHeader(
+            value = "Authorization",
+            required = false
+        ) authorizationHeader: String?
+    ): List<DiscussionPostDTO> {
+        try {
+            authorizationHeader?.let {
+                val user =
+                    jwtService.getUserFromAuthorizationHeader(authorizationHeader) ?: throw Exception("Invalid token")
+                return searchService.searchDiscussionPost(keywords)
+
+            } ?: return searchService.searchDiscussionPost(keywords) //if we want to customize the search results according to user, replace the method call here
+        } catch (e: Exception) {
+            if (e.message == "Invalid token") {
+                throw ResponseStatusException(HttpStatus.UNAUTHORIZED, e.message)
+            } else {
+                throw ResponseStatusException(HttpStatus.BAD_REQUEST, e.message)
+            }
+        }
+    }
+
+    @GetMapping("search_user")
+    fun searchUser(
+        @RequestParam(value = "keywords") keywords: List<String>,
+        @RequestHeader(
+            value = "Authorization",
+            required = false
+        ) authorizationHeader: String?
+    ): List<RegisteredUser> {
```

Comment:
You should return RegisteredUserDTO here

Reply:
Done